cosmwasm/contracts/wormchain-accounting/src/contract.rs

```diff
@@ -0,0 +1,491 @@
+use std::marker::PhantomData;
+
+use accounting::{
+    query_balance, query_modification, query_transfer,
+    state::{account, transfer, Modification, TokenAddress, Transfer},
+};
+use anyhow::{ensure, Context};
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    from_binary, to_binary, Binary, ConversionOverflowError, CosmosMsg, Deps, DepsMut, Empty, Env,
+    Event, MessageInfo, Order, Response, StdError, StdResult, Uint256, WasmMsg,
+};
+use cw2::set_contract_version;
+use cw_storage_plus::Bound;
+use tinyvec::{Array, TinyVec};
+use wormhole::token::Message;
+use wormhole_bindings::WormholeQuery;
+
+use crate::{
+    bail,
+    error::{AnyError, ContractError},
+    msg::{
+        AllAccountsResponse, AllModificationsResponse, AllPendingTransfersResponse,
+        AllTransfersResponse, ExecuteMsg, Instantiate, InstantiateMsg, MigrateMsg, Observation,
+        QueryMsg, Upgrade,
+    },
+    state::{self, Data, PendingTransfer, PENDING_TRANSFERS, TOKENBRIDGE_ADDR},
+};
+
+// version info for migration info
+const CONTRACT_NAME: &str = "crates.io:wormchain-accounting";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut<WormholeQuery>,
+    env: Env,
+    info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, AnyError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)
+        .context("failed to set contract version")?;
+
+    let signatures = msg.signatures.iter().cloned().map(From::from).collect();
+    let _: Empty = deps
+        .querier
+        .query(
+            &WormholeQuery::VerifyQuorum {
+                data: msg.instantiate.clone(),
+                observer_set_index: msg.observer_set_index,
+                signatures,
+                block_time: env.block.height,
+            }
+            .into(),
+        )
+        .context(ContractError::VerifyQuorum)?;
+
+    let init: Instantiate =
```

Comment:
Is this instantiate message similar or the same type of VAA as for deploying other contracts?

Reply:
No, it's a completely different message (defined in src/msg.rs).